backend/dataall/modules/worksheets/services/worksheet_service.py

```diff
@@ -122,3 +150,48 @@ def run_sql_query(session, uri, worksheetUri, sqlQuery):
         )
 
         return AthenaClient.convert_query_output(cursor)
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(RUN_ATHENA_QUERY)
+    @ResourceThresholdService.check_invocation_count('nlq', 'modules.worksheets.features.nlq.max_count_per_day')
+    def run_nlq(uri, prompt, worksheetUri, db_name, table_names):
+        with get_context().db_engine.scoped_session() as session:
+            environment = EnvironmentService.get_environment_by_uri(session, uri)
+            worksheet = WorksheetService._get_worksheet_by_uri(session, worksheetUri)
+
+            env_group = EnvironmentService.get_environment_group(
+                session, worksheet.SamlAdminGroupName, environment.environmentUri
+            )
+
+        glue_client = GlueClient(
+            account_id=environment.AwsAccountId, region=environment.region, role=env_group.environmentIAMRoleArn
+        )
+
```

Comment:
ultra nit: the only metadata that we send are the table names no? I would name the variable table_names instead of metadata so that we know what we are sending as LLM context

Reply:
We send the following for each table selected to the LLM: 
```
            Database Name: {database}
            Table Name: {table_name} 
            Column Metadata: {column_metadata}
            Partition Metadata: {partition_metadata}
```